### PR TITLE
Unit test coverage for Throttling Service Feature (WIP)

### PR DESCRIPTION
**Pull request by NerevarineRule**
## Proposed changes

Work In Progress -
1) Throttling Service Integration test through server-side mocking
2) Throttling Service Component Test

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/IdentityCore.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1631,6 +1631,13 @@
 		E733EDFE25C0A4B100ACB79A /* MSIDThumbprintCalculator.m in Sources */ = {isa = PBXBuildFile; fileRef = E733EDFC25C0A4B100ACB79A /* MSIDThumbprintCalculator.m */; };
```
Comment (kaisong1990) [inline IdentityCore/IdentityCore.xcodeproj/project.pbxproj:1631]: This pull request does not update changelog.txt.

Please consider if this change would be noticeable to a partner or user and either update changelog.txt or resolve this conversation.

File: IdentityCore/src/cache/MSIDKeychainTokenCache.m [hunk 2]
```diff
@@ -763,7 +763,7 @@ - (NSArray *)itemsWithKey:(MSIDCacheKey *)key
     NSData *generic = key.generic;
     NSNumber *type = key.type;
     
-    MSID_LOG_WITH_CTX_PII(MSIDLogLevelVerbose, context, @"Get keychain items, key info (account: %@ service: %@ generic: %@ type: %@, keychainGroup: %@)", MSID_EUII_ONLY_LOG_MASKABLE(account), service, generic, type, [self keychainGroupLoggingName]);
+    MSID_LOG_WITH_CTX_PII(MSIDLogLevelVerbose, context, @"Get keychain items, key info (account: %@ service: %@ generic: %@ type: %@, keychainGroup: %@)", MSID_PII_LOG_MASKABLE(account), service, generic, type, [self keychainGroupLoggingName]);
```
Comment (oldalton) [inline IdentityCore/src/cache/MSIDKeychainTokenCache.m:766]: Can you double check why this PR is reverting the EUII logging change?

**Review by oldalton — CHANGES_REQUESTED** [review]

File: IdentityCore/tests/MSIDLRUCacheTest.m [hunk 3]
```diff
@@ -336,7 +337,7 @@ - (void)testMSIDLRUCache_whenCallingAPIsUseThrottlingCacheWithinGCDBlocks_thrott
                 XCTAssertNil(subError);
 
                 MSIDThrottlingCacheRecord *throttleCacheRecord = [[MSIDThrottlingCacheRecord alloc] initWithErrorResponse:nil
-                                                                                                             throttleType:cacheKeyFromOtherQueue
+                                                                                                             throttleType:(i + 50)
```
[the remark this answers is not in the document]
Comment (NerevarineRule) [reply IdentityCore/tests/MSIDLRUCacheTest.m:340]: Two APIs are trying to modify the same cache entry, mapped by throttleType value. one is trying to add it, while the other is trying to remove it. first loop is from 0 to 49, while the second loop is from 50 to 99.

File: IdentityCore/src/cache/MSIDKeychainTokenCache.m (hunk 2 above)
Comment (NerevarineRule) [reply IdentityCore/src/cache/MSIDKeychainTokenCache.m]: git merge dev didn't work. somehow ended up reverting all the log changes when I tried to merge with dev and then with Hieu's branch. something went wrong. I had to update this branch to have all of your logging changes manually.

**Pull request merged**